src/concrete/ml/sklearn/base.py

```diff
@@ -651,41 +799,59 @@ def predict(self, X: Data, fhe: Union[FheMode, str] = FheMode.DISABLE) -> numpy.
         # Quantize the input
         q_X = self.quantize_input(X)
 
+        predict_method: Callable
         # If the inference is executed in FHE or simulation mode
         if fhe in ["simulate", "execute"]:
             # Check that the model is properly compiled
             self.check_model_is_compiled()
 
-            q_y_pred_list = []
-            for q_X_i in q_X:
-                # Expected encrypt_run_decrypt output shape is (1, n_features) while q_X_i
-                # is of shape (n_features,)
-                q_X_i = numpy.expand_dims(q_X_i, 0)
+            # For mypy, even though we already check this with self.check_model_is_compiled()
+            assert self.fhe_circuit is not None
+
+            # If the inference should be executed using simulation
+            if fhe == "simulate":
+                if self._ciphertext_format == CiphertextFormat.TFHE_RS:
+                    raise ValueError("Can't simulate with TFHE-rs cts")
 
-                # For mypy, even though we already check this with self.check_model_is_compiled()
-                assert self.fhe_circuit is not None
+                is_crt_encoding = self.fhe_circuit.statistics["packing_key_switch_count"] != 0
 
-                # If the inference should be executed using simulation
-                if fhe == "simulate":
-                    is_crt_encoding = self.fhe_circuit.statistics["packing_key_switch_count"] != 0
+                # If the virtual library method should be used
+                # For now, use the virtual library when simulating
+                # circuits that use CRT  encoding because the official simulation is too slow
+                # FIXME: https://github.com/zama-ai/concrete-ml-internal/issues/4391
+                if USE_OLD_VL or is_crt_encoding:
```

Comment:
I didn't know USE_OLD_VL what still there

Reply:
it's disabled but still there

Reply:
Old vl is simply CP graph execution, so it is tested in Concrete. but you're right in the sense that we could easily remove it - we kept it thinking it might be useful at some point, but it's not anymore

Reply:
> it is tested in Concrete
I don't know if they still test it. At some point, they'll certainly remove it as well